ngmt/modules/ptd/_pham.py

```diff
@@ -0,0 +1,484 @@
+            dt_data (pd.Series, optional): Original datetime in the input data. If original datetime is provided, the output onset will be based on that.
+            tracking_system (str, optional): Tracking systems.
+            tracked_point (str, optional): Tracked points on the body.
+            plot_results (bool, optional): If True, generates a plot. Default is False.
+
+        Returns:
+            The postural transition information is stored in the 'postural_transitions_' attribute,
+            which is a pandas DataFrame in BIDS format with the following columns:
+                - onset: Start time of the postural transition in second.
+                - duration: Duration of the postural transition in second.
+                - event_type: Type of the event which is postural transition.
+                - tracking_systems: Name of the tracking systems.
+                - tracked_points: Name of the tracked points on the body.
+        """
+        # check if dt_data is a pandas Series with datetime values
+        if dt_data is not None and (
+            not isinstance(dt_data, pd.Series)
+            or not pd.api.types.is_datetime64_any_dtype(dt_data)
+        ):
+            raise ValueError("dt_data must be a pandas Series with datetime values")
+
+        # check if dt_data is provided and if it is a series with the same length as data
+        if dt_data is not None and len(dt_data) != len(data):
+            raise ValueError("dt_data must be a series with the same length as data")
+
+        # Check if data is a DataFrame
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError("Input data must be a pandas DataFrame")
+        
+        # Error handling for invalid input data
+        if not isinstance(data, pd.DataFrame) or data.shape[1] != 6:
+            raise ValueError(
+                "Input accelerometer and gyro data must be a DataFrame with 6 columns for x, y, and z axes."
+            )
+
+        # Check if sampling frequency is positive
+        if sampling_freq_Hz <= 0:
+            raise ValueError("Sampling frequency must be positive")
+
+        # Check if plot_results is a boolean
+        if not isinstance(plot_results, bool):
+            raise ValueError("plot_results must be a boolean value")
+
+        # Calculate sampling period
+        sampling_period = 1 / sampling_freq_Hz
+
+        # Select acceleration data and convert it to numpy array format
+        accel = data.iloc[:, 0:3].copy()
+        accel = accel.to_numpy()
+
+        # Select gyro data and convert it to numpy array format
+        gyro = data.iloc[:, 3:6].copy()
+        self.gyro = gyro.to_numpy()
+
+        # Extract vertical accelerometer data using the specified index
+        self.gyro_mediolateral = data[gyro_mediolateral].to_numpy()
+
+        # Check unit of acceleration data if it is in g or m/s^2 (including variations)
+        if accel_unit in ["m/s^2", "meters/s^2", "meter/s^2"]:
```

Comment:
Check allowed units in BIDS, maybe we have to enforce a way how to write Units in NGMT

Reply:
These parts of the codes were updated. With this update, all merge comments were already handled.